packages/engine-core/src/library/DefaultLibraryLoader.ts

```diff
@@ -0,0 +1,184 @@
+import Debug from '@prisma/debug'
+import { getEnginesPath } from '@prisma/engines'
+import { getNodeAPIName, getPlatform, Platform } from '@prisma/get-platform'
+import chalk from 'chalk'
+import fs from 'fs'
+import path from 'path'
+
+import { EngineConfig } from '../common/Engine'
+import { PrismaClientInitializationError } from '../common/errors/PrismaClientInitializationError'
+import { printGeneratorConfig } from '../common/utils/printGeneratorConfig'
+import { fixBinaryTargets } from '../common/utils/util'
+import { Library, LibraryLoader } from './types/Library'
+
+const debug = Debug('prisma:client:libraryEngine:loader')
+
+export class DefaultLibraryLoader implements LibraryLoader {
+  private config: EngineConfig
+  private libQueryEnginePath: string | null = null
+  private platform: Platform | null = null
+
+  constructor(config: EngineConfig) {
+    this.config = config
+  }
+
+  async loadLibrary(): Promise<Library> {
```

Comment:
Most of this file is moved from `LibraryEngine` class as-is, including comments and TODOs